server/xpub/entities/file/index.js

```diff
@@ -23,12 +23,16 @@ const FileManager = {
   delete: dataAccess.delete,
   new: (props = {}) => lodash.merge({}, empty, props),
   save: async file => {
+    let id = { file }
     if (file.id) {
-      await dataAccess.update(file)
-      return file
+      const updated = await dataAccess.update(file)
+      if (!updated) {
+        throw new Error('File not found')
```

Comment:
Does this throw when the item is identical and so not updated? Is this desired behaviour?

Reply:
Ah no, sorry bad variable name. `updated` is the number of rows that were updated.

Reply:
ok ... so if there is nothing to update because all the values are the same does this still throw? I assume it would as `updated == 0`